app/views/allnewsview.py

```diff
@@ -0,0 +1,148 @@
+
+
+#   Model required by flask_restx for expect on POST and PUT methods
+model_validator = local_ns.model(CURRENT_NAME, {
+    'source': fields.String,
+    'author': fields.String,
+    'title': fields.String,
+    'description': fields.String,
+    'url' : fields.String,
+    'urlToImage':  fields.String,
+    'publishedAt': fields.String,
+    'content': fields.String
+})
+
+@local_ns.route('/')
+class AllNewsList(Resource):
+    @local_ns.doc('Get all the' +CURRENT_NAME+ 's')
+    def get(self):
+        try:
+            if isOnDev:
+                response = jsonify(TheModel.find_all())
+                response.status_code = HttpStatus.OK
+
+        except Exception as e:
+            response = jsonify({'message': e.__str__()})
+            response.status_code = HttpStatus.INTERNAL_ERROR
+        return response
+
+    @local_ns.doc('Create an ' + CURRENT_NAME)
+    @local_ns.expect(model_validator)
+    def post(self):
+        if not isOnDev:
+            response = jsonify({'message': 'Not allowed'})
+            response.status_code = HttpStatus.NOT_ALLOWED
+            return response
+        try:
+            element_json = request.get_json()
+            element_data = allnews_schema.load(element_json)
+            element_data.save_to_db()
+            response = jsonify(element_data.json())
+            response.status_code = HttpStatus.CREATED
+        except Exception as e:
+            response = jsonify({'message': e.__str__()})
+            response.status_code = HttpStatus.BAD_REQUEST
+        return response
+
+
+@local_ns.route('/<int:id>')
+class AllNews(Resource):
+    @local_ns.doc('Get the ' + CURRENT_NAME + ' with the specified id',
+                  params={'id': 'id of the ' + CURRENT_NAME + ' to get'})
+    def get(self, id):
+        try:
+            element_data = TheModel.find_by_id(id)
+            if element_data:
+                response = jsonify(element_data.json())
+                response.status_code = HttpStatus.OK
+            else:
+                response = jsonify({'message': CURRENT_NAME + ' not found.'})
+                response.status_code = HttpStatus.NOT_FOUND
+        except Exception as e:
+            response = jsonify({'message': e.__str__()})
+            response.status_code = HttpStatus.INTERNAL_ERROR
+        return response
+
+    @local_ns.doc('Update an ' + CURRENT_NAME + ' with the specified id',
+                  params={'id': 'id of the ' + CURRENT_NAME + ' to update'})
+    @local_ns.expect(model_validator)
+    def put(self, id):
+        if not isOnDev:
+            response = jsonify({'message': 'Not allowed'})
+            response.status_code = HttpStatus.NOT_ALLOWED
+            return response
+        try:
+            element_data = TheModel.find_by_id(id)
+
+            if element_data:
+                element_data.source = EmptyValues.EMPTY_STRING if request.json['source'] == EmptyValues.EMPTY_STRING else request.json['source']
```

Comment:
Could you extract all this heavy logic someplace else? I find that the views in general should be light and contain little logic (yes, I am aware mappings are not complex logic per se but it does make the view much "fatter" and harder to read)